### ISSUE TITLE
Circular buffer

### ISSUE THREAD

**Issue by altavir** [lamba92/kotlin-extlib#6]
A fixed size buffer, where you can add a new element automatically evicting last one (implemented by fixed size array with moving begin pointer).
In Java I would use `ArrayDeque` with manual adding and popping, but it is not very convenient and not thread safe.

**Comment by lamba92** [comment]
Well for concurrency you could always use AtomicFU for whatever object and make accessing it atomic.

**Comment by altavir** [comment]
Yep. It is not that easy to use though. I will implement it in kmath and if it works like I want, I will move it here.
